src/Features/Core/Portable/Completion/Providers/AbstractObjectInitializerCompletionProvider.cs

```diff
@@ -62,8 +62,9 @@ internal abstract class AbstractObjectInitializerCompletionProvider : CommonComp
 
             foreach (var uninitializedMember in uninitializedMembers)
             {
+                var displayText = uninitializedMember.GetParameters().Length == 0 ? uninitializedMember.ToNameDisplayString() : uninitializedMember.Name;
```

Comment:
i don't think we want ToNameDisplayString at all.
Something is causing the insertion codepath to not go through CompletionUtilities.GetInsertionText (which would properly escape the string as necessary).

Reply:
The problem here seems to be that AbstractObjectINitializerCompletionProvider does not do the same things as AbstractSymbolCompletionProvider where it asks what the appropriate 'insertion' text should be.

Reply:
@CyrusNajmabadi I didn't fully understand your comment, but wanted to mention that  in the completion for patterns (which had the same problem with escaping keywords), I used `untestedMember.Name.EscapeIdentifier()`. Would that be appropriate here as well?

See https://github.com/dotnet/roslyn/pull/26553/files#diff-71f1153caf63996a37515baedd143adbR53

Reply:
looking at this code overall... yes, i think that would be sufficient.